test/fixtures/person.rb

```diff
@@ -13,3 +13,17 @@ class ProfileData < ActiveResource::Base
     self.site = "http://external.profile.data.nl"
   end
 end
+
+module Camelcase
+  class Person < ::Person
+    def load(attributes, *args)
+      attributes = attributes.deep_transform_keys { |key| key.to_s.underscore }
+
+      super
+    end
+
+    def serializable_hash(options = {})
+      super.deep_transform_keys! { |key| key.camelcase(:lower) }
+    end
```

Comment:
Is this the best way to achieve this outcome? When learning about the library, my first instinct was to tackle name casing at the `JsonFormat` level.

Unfortunately, since `#encode` is implemented in terms of `#to_json`, and not `JsonFormat.encode`, a custom format that inherited from `JsonFormat` had no effect.

Is there a more canonical way to extend or configure the pre-existing `JsonFormat` module to more elegantly handle this?

Could it be worthwhile to explore changing the `Base#encode` implementation to incorporate `self.class.format.encode`?